src/coreclr/jit/valuenum.cpp

```diff
@@ -1881,12 +1885,14 @@ ValueNum ValueNumStore::VNForSimd64Con(const simd64_t& cnsVal)
 {
     return VnForConst(cnsVal, GetSimd64CnsMap(), TYP_SIMD64);
 }
+#endif // TARGET_XARCH
 
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
```

Comment:
You should use `#if defined(FEATURE_MASKED_HW_INTRINSICS)` instead instead of `defined(TARGET_XARCH) || defined(TARGET_ARM64)`